### PR TITLE
Editor: remove overlay from EditorConfirmationSidebar

### PR DESCRIPTION
**Pull request by michaeldcain**
This PR allows the user to continue to proofread their changes while the `EditorConfirmationSidebar` is visible. The sidebar will be hidden if the user makes any edits to their post.

**Mock:**
screencapture_at_tue_jun_13_09_44_09_edt_2017

This PR is part of a larger epic. The feature is behind a feature-flag and is intentionally incomplete. See p8F9tW-3F-p2.

**To Test:**
- `ENABLE_FEATURES=post-editor/delta-post-publish-flow make run`
- Draft a post.
- Click "Publish…" button.
- When confirmation sidebar is visible, post should remain visible and editable (no overlay), and be shifted to fit beside sidebar
- When post is edited, confirmation sidebar should close and post settings sidebar should follow whether or not it was displayed before "Publish…" was clicked.

### CODE REVIEW

File: client/post-editor/editor-publish-button/index.jsx [hunk 1]
```diff
@@ -92,13 +92,20 @@ export class EditorPublishButton extends Component {
...
+					return this.props.translate( 'Publish' );
 				}
-				return this.props.translate( 'Publish' );
+
+				if ( this.props.isPublishing ) {
+					return this.props.translate( 'Publishing…',
+						{ comment: 'Button label on the editor sidebar while publishing is in progress' } );
+				}
+
+				return this.props.translate( 'Publish…',
```
Comment (a8ci18n) [inline client/post-editor/editor-publish-button/index.jsx]: Hi! I've found a possible matching string that has already been translated **104** times:
`translate( 'Publish' )` *ES Score: 15*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/remove/editor-confirmation-sidebar-overlay)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/post-editor/editor-publish-button/index.jsx [hunk 2]
```diff
@@ -92,13 +92,20 @@ export class EditorPublishButton extends Component {
...
-				if ( this.props.isConfirmationSidebarEnabled ) {
-					return this.props.translate( 'Publish…',
-						{ context: 'Button label on the editor sidebar - a confirmation step will follow' } );
+				if ( ! this.props.isConfirmationSidebarEnabled ) {
+					return this.props.translate( 'Publish' );
 				}
-				return this.props.translate( 'Publish' );
+
+				if ( this.props.isPublishing ) {
+					return this.props.translate( 'Publishing…',
```
Comment (a8ci18n) [inline client/post-editor/editor-publish-button/index.jsx]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Publishing...' )` *ES Score: 20*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/remove/editor-confirmation-sidebar-overlay)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by mattsherman** [comment]
One issue I've found so far... if you have the confirmation sidebar open and click on the post title, then click on the post body, the confirmation sidebar hides, even though no change has been made to the post.

confirmation sidebar hides

**Comment by mattsherman** [comment]
Publishing button positioning is off on mobile:

screencapture at tue jul 11 17 57 04 edt 2017

**Comment by jonathansadowski** [comment]
@mattsherman I pushed some changes that should address the issues you noticed.

**Comment by mattsherman** [comment]
Changes look good to me! 👍 Tested desktop and mobile, and also without flag enabled.

**Pull request merged**